src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs

```diff
@@ -157,7 +157,25 @@ private Task<Document> UpdateDocumentAsync(Document document, SyntaxNode root, S
                 }
                 else
                 {
-                    content.Add(generator.Interpolation(piece.WithoutTrivia()));
+                    var firstInterpolationToken = piece.GetFirstToken();
+                    var lastInterpolationToken = piece.GetLastToken();
+
+                    if (firstInterpolationToken.RawKind == startToken.RawKind && lastInterpolationToken.RawKind == endToken.RawKind)
+                    {
+                        // If the piece like $"foo {a} goo: ",
+                        // we get the sub string between the first two caracters (whitch represent intepolation startToken)
+                        // and the last one that represent interpolation endToken 
+                        // objective here is to transform the non-StringOrCharacterLiteral to a string literal
+                        var pieceString = piece.ToString();
+                        var textWithoutInterpolationTokens = pieceString.Substring(2, ((pieceString.Length - 1) - 2));
```

Comment:
```suggestion
                        var textWithoutInterpolationTokens = pieceString[2..^1];
```

Reply:
what is the '2' for?

Reply:
2 is to determinate the correct value of the length parameter, because the substring length equal string length minus the number of characters we need to skip.